src/Utility/Queue.hpp

```diff
@@ -0,0 +1,114 @@
+#pragma once
+
+namespace Utility {
+    /// Queue container class.
+    template<typename T> class Queue {
+        public:
+            /// Constructor.
+            Queue() = default;
+
+            /// Push value at front of queue.
+            /**
+             * @param val Value to push.
+             */
+            void Push(const T& val);
+
+            /// Get value at front of queue.
+            /**
+             * @return Value at front of queue.
+             */
+            T* Front();
+
+            /// Get value at back of queue.
+            /**
+             * @return Value at back of queue.
+             */
+            T* Back();
+
+            /// Remoce value at front of queue.
```

Comment:
Spelling error

Shouldn't a queue push and pop things at opposite ends of the queue? Otherwise it's a stack. The documentation says both work at the front of the queue.